pkg/kev/services.go

```diff
@@ -134,9 +134,52 @@ func (sc ServiceConfig) validate() error {
 		return err
 	}
 
-	if !result.Valid() {
-		return errors.New(result.Errors()[0].Description())
+	if result.Valid() {
+		return nil
 	}
+
+	// Prioritise clear error messages.
+	if e := findError(result, withType("required")); e != nil {
+		return errors.New(e.Description())
+	}
+
+	// Exclude errors that are very cryptic and hurt usability.
+	if e := findError(result, excludeTypes("number_one_of", "number_any_of", "number_all_of")); e != nil {
+		return errors.New(e.Description())
+	}
+
+	// If we don't find anything useful just go with whatever is available.
+	return errors.New(result.Errors()[0].Description())
+}
+
+func excludeTypes(ts ...string) func(gojsonschema.ResultError) bool {
+	return func(re gojsonschema.ResultError) bool {
+		for _, t := range ts {
+			if t == re.Type() {
+				return true
```

Comment:
as you exclude I guess this should return false, and true by default otherwise?

Reply:
I agree with @marcinc here.

Reply:
Otherwise, it makes sense. 👌

As a final nice to have, it would be great if we can move the schema logic into a separate file if possible.

Reply:
yep, this is why the build was failing, it's fixed now. Do you think this is more clear/easier to understand without a lot of context?

Reply:
also I agree with this not being the right place for this but is there a good place? Is it going to be re-used somewhere else? I wonder if this would make sense as a contribution to the upstream repo